src/types/Family.ts

```diff
@@ -180,7 +181,16 @@ export class CurrentFamily {
 
   public async inviteMembers(memberEmails: string[]) {
     const currentFamilyDoc = await this.currentFamilyDoc();
-    await currentFamilyDoc.update('pendingMembers', FieldValue.arrayUnion(...memberEmails));
+    const currentPendingMembers = (await currentFamilyDoc.get()).get('pendingMembers') ?? [];
+    const currentPendingMembersEmails = currentPendingMembers.map(pendingMember => pendingMember.email);
+
+    const membersToBeInvited = memberEmails.filter(email => !currentPendingMembersEmails.includes(email));
+
+    const memberEmailsWithDates = membersToBeInvited.map(memberEmail => ({
+      email: memberEmail,
+      date: firebase.firestore.Timestamp.now()
+    }));
+    await currentFamilyDoc.update('pendingMembers', FieldValue.arrayUnion(...memberEmailsWithDates));
```

Comment:
Would this be a valid refactor?
```suggestion
    const currentPendingMembers = (await currentFamilyDoc.get()).get('pendingMembers') ?? [];

    const pendingMembers = memberEmails.map(memberEmail => ({
      email: memberEmail,
      date: currentPendingMembers.find(m => m.email === memberEmail)?.date ?? firebase.firestore.Timestamp.now()
    }));
    await currentFamilyDoc.update('pendingMembers', pendingMembers);
```

Reply:
Nope, this code re-writes already existing pending members' emails.